### DISCUSSION TITLE
Usage of asyncio.wait_for

### DISCUSSION THREAD

**Discussion by krishnak** [micropython/micropython#9511] [Using MicroPython] [answered]
I have two esp 32 boards  running micropython.

On Board A an interrupt occurs
the interrupt starts a asyncio task like this and returns
`asyncio.run(self.delayedEvent(eventNumber,False))
`
the delayedEvent sends a message to Board B  and waits for a response
Board B may respond or may not respond.

I have an asyncio task in Board A waiting for responses from B which adds them to a Queue 

It sets a responseEvent - asnycio event

I want to wait on the event but since the event may not even occur, I have to time out, Hence I tried
```

                asyncio.wait_for(responseQueueEvent.wait(),1)        
               or                  asyncio.wait_for(responseQueueEvent,1) 
```       
the above doesn't wait at all, can you please correct the syntax or my misinterpretation that this is how this is done!

**Comment by peterhinch** [comment, marked as answer]
This is an odd use of `asyncio.run`. The best way to interface between interrupts and `uasyncio` is to use [ThreadSafeFlag](http://docs.micropython.org/en/latest/library/uasyncio.html#uasyncio.ThreadSafeFlag); see also [this ref](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md#36-threadsafeflag). You might also want to read [these notes](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md#522-tasks-with-timeouts) on the use of timeouts - it looks like you're missing an `await`.

**Reply by krishnak** [reply]
Thanks, while waiting for an answer, I did venture in to that route i.e to use ThreadSafeFlag - now it is better. Thank you for your tutorial, I have raised a question as an issue on your Queue implementation in your repository. I look forward to your input on that too.

**Reply by peterhinch** [reply]
I think you should start from square one with `uasyncio`, working through examples in [the tutorial](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md) so that you learn the basic concepts. It's an easier route than trial and error.
